src/spikeinterface/comparison/comparisontools.py

```diff
@@ -707,7 +707,7 @@ def do_confusion_matrix(event_counts1, event_counts2, match_12, match_event_coun
     N2 = len(unit2_ids)
 
     matched_units1 = match_12[match_12 != -1].index
-    matched_units2 = match_12[match_12 != -1].values
+    matched_units2 = match_12[match_12 != -1].values.astype(int)
```

Comment:
Is it unit_ids ?
If yes we neeed to handle the str case.

Reply:
Yes. I suggest we do it in a different PR